README.md

```diff
@@ -43,6 +45,8 @@ To work with this datasource the backend needs to implement 3 endpoints:
 
 - `GET /` with 200 status code response. Used for "Test connection" on the datasource config page.
 - `POST /search` to return available metrics.
```

Comment:
Can we remove `/search` in favor of `/metrics`?

Reply:
The "/search" API is reserved for compatibility with code mode. If the "/search" API is deleted, it means that previous users need to modify the code.

I did not use the "/search" API because I felt that the name of "/search" did not match the actual behavior. Neither the code mode (original) nor the builder mode searched for metrics through this API.

Reply:
@MicroOps-cn yup, I'd deprecate `/search` in v0.5 and we would drop it in v0.6 where this feature would land.

Basically those seems the same except that `/metrics` has additional options. So basically `/metrics` should work for the current mode just like that.

Reply:
Do you plan to delete the code mode, or just replace the "/search" API with "/metrics"?

If the "/search" API is replaced by "/metrics", it is better to make some compatibility in the code, so as to ensure that the original user can migrate to the new API more easily.

Reply:
I have done compatibility processing on the `listMetrics` API. If you want to delete the "/search" API, you only need to change the `searchMetrics` in the `src/component/queryeditor.tsx` file to `listMetrics`.

Reply:
As for when to remove the `/search` API, I think it's better for you to decide.

Reply:
Ok. I'm fine with doing BC breaks between versions since it makes maintenance easier and leads to better datasource faster. Though I intend to specify upgrade path in the release notes for sure.